ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/source/BlobSourceInfo.kt

```diff
@@ -90,5 +92,114 @@ class BlobSourceInfo : AbstractSourceInfo {
             blobUploadContainer: BlobUploadContainer,
         ): BlobSourceInfo =
             fromLocalSource(streamSourceInfo, blobUploadContainer)
+
+        // batch convert multiple LocalSource objects to BlobSourceInfo using parallel uploads
+        suspend fun fromLocalSourcesBatch(
+            localSources: List<LocalSource>,
+            blobUploadContainer: BlobUploadContainer,
+        ): BatchConversionResult {
+            if (localSources.isEmpty()) {
+                return BatchConversionResult(emptyList(), emptyList())
+            }
+
+            logger.info(
+                "Starting batch conversion of {} local sources",
+                localSources.size,
+            )
+
+            val uploadSources =
+                localSources.map { source ->
+                    val (inputStream, size, effectiveCompression) =
+                        source.prepareForUpload()
+                    val blobName = source.generateBlobName()
+                    UploadSource(
+                        name = blobName,
+                        stream = inputStream,
+                        sizeBytes = size ?: -1,
+                    )
+                }
+
+            val uploadResults =
+                blobUploadContainer.uploadManyAsync(uploadSources)
```

Comment:
Multi upload tests with a format combination as a test perhaps. Is that a supported scenario

Reply:
this is tricky, we dont have any validation before uploading to storage, but I think the format validation happens later during the ingestion phase when submitting to Kusto, in QueuedIngestionClient.submitQueuedIngestion(), there input parameters we have format, where a single IngestRequest is created with one format for all blobs. This is an interesting scenario, a testcase for the same as suggested will defintely help.